play-validations/memory-footprint/src/main/java/com/google/wear/watchface/dfx/memory/AmbientMemoryFootprintCalculator.java

```diff
@@ -135,90 +152,32 @@ long computeAmbientMemoryFootprint(long screenWidth, long screenHeight) {
-                    }
-                }
-
-                sets = combinedSets;
-            }
-
-            // Compute the maximum memory needed by any set of resources.
-            long maxTotal = 0;
-            int numResources = 0;
-            long maxResourceSize = 0;
-            for (Set<String> set : sets) {
-                long total = 0;
-                for (String resource : set) {
-                    DrawableResourceDetails details =
-                            DrawableResourceDetails.findInMap(resourceMemoryMap, resource);
-                    long imageBytes = details.getBiggestFrameFootprintBytes();
-
-                    // If this image can be downsampled then the size is halved.
-                    if (details.canUseRGB565()) {
-                        imageBytes /= 2;
-                    }
-
-                    if (evaluationSettings.isVerbose()) {
-                        System.out.printf(
-                                "Counting resource %s; %s bytes, %s mb, %s x %s %s%n",
-                                resource,
-                                details.getBiggestFrameFootprintBytes(),
-                                ((double) imageBytes) / 1024 / 1024,
-                                details.getWidth(),
-                                details.getHeight(),
-                                details.canUseRGB565() ? "RGB565" : "ARGB8888");
-                    }
-                    total += imageBytes;
-                    if (maxResourceSize < imageBytes) {
-                        maxResourceSize = imageBytes;
-                    }
-                }
-                if (maxTotal < total) {
-                    maxTotal = total;
-                }
-                if (numResources < set.size()) {
-                    numResources = set.size();
-                }
-            }
-
-            if (evaluationSettings.isVerbose()) {
-                long average = (numResources > 0) ? (maxTotal / numResources) : 0;
-                System.out.printf(
-                        "Resource count %s average size %s max size %s\n",
-                        numResources, average, maxResourceSize);
-            }
-
-            return maxTotal;
+        if (evaluationSettings.isVerbose()) {
+            System.out.printf(
+                    "Counting resource %s; %s bytes, %s mb, %s x %s %s%n",
+                    resource,
+                    details.getBiggestFrameFootprintBytes(),
+                    ((double) imageBytes) / 1024 / 1024,
+                    details.getWidth(),
+                    details.getHeight(),
+                    details.canUseRGB565() ? "RGB565" : "ARGB8888");
         }
+        return imageBytes;
     }
 
     private class Visitor {
-        private final PerConfigurationDynamicResources perConfigurationDynamicResources;
+        private final DrawableNodeConfigTable drawableNodeConfigTable =
+                new DrawableNodeConfigTable();
```

Comment:
Instead of computing the user config permutations, we collect the UserConfigSet, meaning the pairings of user config keys and values, for each node that is drawn dynamically ie Clock, ComplicationSlot etc